src/Publisher.ts

```diff
@@ -951,17 +957,16 @@ export default class Publisher {
 							? `$<div class="markdown-embed-title">\n\n${header}\n\n</div>\n`
 							: "";
 						let embedded_link = "";
-						if (
+						const publishedFilesContainsLinkedFile =
 							publishedFiles.find(
 								(f) => f.path == linkedFile.path,
-							)
-						) {
-							const gardenPath = metadata.frontmatter[
-								"dg-permalink"
-							]
-								? sanitizePermalink(
-										metadata.frontmatter["dg-permalink"],
-								  )
+							);
+						if (publishedFilesContainsLinkedFile) {
+							const permalink =
+								metadata?.frontmatter &&
+								metadata.frontmatter["dg-permalink"];
+							const gardenPath = permalink
+								? sanitizePermalink(permalink)
```

Comment:
for stuff like this some functional language like ramda / lodash / or some typescript optimized alternative would make things like this much nicer! 


`const getPermalink = R.path(["frontmatter", PERMALINK_FRONTMATTER_KEY])` or whatever

in code 

```
const permalink = getPermalink(frontmatter) 
```

undefineds are dealt with automatically and code is legible!